.github/workflows/ci.yml

```
@@ -192,135 +192,107 @@ jobs:
 
       - name: Test diesel-derives (nightly)
         if: matrix.rust == 'nightly'
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path diesel_derives/Cargo.toml --no-default-features --features "diesel/${{ matrix.backend }} diesel/unstable"
+        shell: bash
+        run: cargo +${{ matrix.rust }} test --manifest-path diesel_derives/Cargo.toml --no-default-features --features "diesel/${{ matrix.backend }} diesel/unstable"
 
       - name: Test diesel-derives
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path diesel_derives/Cargo.toml --no-default-features --features "diesel/${{ matrix.backend }}"
+        shell: bash
+        run: cargo +${{ matrix.rust }} test --manifest-path diesel_derives/Cargo.toml --no-default-features --features "diesel/${{ matrix.backend }}"
 
       - name: Test diesel-cli
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path diesel_cli/Cargo.toml --no-default-features --features "${{ matrix.backend }}"
+        shell: bash
+        run: cargo +${{ matrix.rust }} test --manifest-path diesel_cli/Cargo.toml --no-default-features --features "${{ matrix.backend }}"
 
       - name: Test diesel examples
         shell: bash
+        env:
+          BACKEND: ${{ matrix.backend }}
         run: |
-          (cd examples/${{ matrix.backend }} && BACKEND=${{ matrix.backend }} ./test_all)
+           (cd examples/${{ matrix.backend}} && rustup run ${{ matrix.rust}} bash test_all)
 
       - name: Test migrations-internals
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path diesel_migrations/migrations_internals/Cargo.toml
+        shell: bash
+        run: cargo +${{ matrix.rust }} test --manifest-path diesel_migrations/migrations_internals/Cargo.toml
 
       - name: Test migrations-macros
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path diesel_migrations/migrations_macros/Cargo.toml --features "diesel/${{ matrix.backend }} ${{ matrix.backend }}"
+        shell: bash
+        run: cargo +${{ matrix.rust }} test --manifest-path diesel_migrations/migrations_macros/Cargo.toml --features "diesel/${{ matrix.backend }} ${{ matrix.backend }}"
 
       - name: Test diesel_migrations
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --manifest-path diesel_migrations/Cargo.toml --features "${{ matrix.backend }} diesel/${{ matrix.backend }}"
+        shell: bash
+        run: cargo +${{ matrix.rust }} test --manifest-path diesel_migrations/Cargo.toml --features "${{ matrix.backend }} diesel/${{ matrix.backend }}"
 
       - name: Run diesel_tests (nightly)
         if: matrix.run == 'nightly'
```

Pre-existing but this should be `matrix.rust`?

```suggestion
        if: matrix.rust == 'nightly'
```